### PR TITLE
GDPR-51 Starting to handle the various event types

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/datacompliance/config/WebClientConfig.java [hunk 1]
```diff
@@ -45,6 +46,9 @@ WebClient authorizedWebClient(final OAuth2AuthorizedClientManager authorizedClie
 
         return WebClient.builder()
                 .apply(oauth2Client.oauth2Configuration())
+                .exchangeStrategies(ExchangeStrategies.builder()
```
Comment (brightonsbox) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/config/WebClientConfig.java:49]: Now in line with some of our other WebClient configuration.

File: src/main/java/uk/gov/justice/hmpps/datacompliance/events/listeners/OffenderDeletionListener.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    private static final String OFFENDER_PENDING_DELETION_REFERRAL_COMPLETE_EVENT = "DATA_COMPLIANCE_OFFENDER-PENDING-DELETION-REFERRAL-COMPLETE";
+    private static final String OFFENDER_DELETION_COMPLETE_EVENT = "DATA_COMPLIANCE_OFFENDER-DELETION-COMPLETE";
+
+    private final Map<String, MessageHandler> messageHandlers = Map.of(
+            OFFENDER_PENDING_DELETION_EVENT, this::handlePendingDeletion,
+            OFFENDER_PENDING_DELETION_REFERRAL_COMPLETE_EVENT, this::handleReferralComplete,
+            OFFENDER_DELETION_COMPLETE_EVENT, this::handleDeletionComplete);
+
+    private final ObjectMapper objectMapper;
+    private DeletionReferralService deletionReferralService;
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/events/listeners/OffenderDeletionListener.java:36]: meh, inconsistent final-ling

File: src/main/java/uk/gov/justice/hmpps/datacompliance/services/DeletionReferralService.java [hunk 3]
```diff
@@ -14,8 +17,26 @@
...
     private OffenderDeletionReferralRepository offenderDeletionReferralRepository;
+    private OffenderDeletionGrantedEventPusher offenderDeletionGrantedEventPusher;
 
-    public void storeOffenderDeletionReferral(final OffenderPendingDeletionEvent event) {
+    public void handlePendingDeletion(final OffenderPendingDeletionEvent event) {
+        storeOffenderDeletionReferral(event);
+
+        // TODO GDPR-51 Complete retention checks on offender record
+
+        offenderDeletionGrantedEventPusher.sendEvent(event.getOffenderIdDisplay());
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/services/DeletionReferralService.java:27]: no danger of this actually triggering detenions of stuff?

Comment (brightonsbox) [reply src/main/java/uk/gov/justice/hmpps/datacompliance/services/DeletionReferralService.java:27]: It's not deployed to production, and there's nothing kicking the whole process off so pretty safe

**Pull request merged**